### DISCUSSION TITLE
Use custom swc transform

### DISCUSSION THREAD

**Discussion by uonr** [vercel/next.js#35502] [Ideas]
### Describe the feature you'd like to request

[FormatJS provides the SWC plugin](https://formatjs.io/docs/tooling/swc-plugin/), but there is no way to integrate the SWC plugin into next.js.

[related comments in next.js discussion](https://github.com/vercel/next.js/discussions/30174#discussioncomment-1859249)

### Describe the solution you'd like

add a field in `next.config.js` to apply custom swc plugins.

### Describe alternatives you've considered

integrate formatjs swc plugin into next.js

**Comment by joelstein** [comment]
I would also like to use FormatJS (react-intl) in Next.js with the SWC plugin.

Is this possible through configuration?

Or is this only possible within some internal component of Next.js?

**Comment by markhughes** [comment]
It looks like the only current way is to hack it in using something like [patch-package](https://www.npmjs.com/package/patch-package)?

+ updating package.json and recompiling next, to force a bump in the swc version?

**Comment by joelstein** [comment]
@markhughes Thanks for that suggestion. Do you have example code that's working for you?

**Reply by markhughes** [reply]
Sorry, my "hack" doesn't allow using so nicely with next, nor allow using the plugins so easily. I'm still studying the swc implementation by next.

You can hack them into here:
https://github.com/vercel/next.js/blob/16f7084e7fa1fb798e68dfb9470896d4258f6492/packages/next-swc/crates/core/src/lib.rs#L132-L204

and recompile. 

But I don't think this is a solution

**Comment by chendong-xie** [comment]
@uonr any update? tks
